### ISSUE TITLE
PROPOSAL - Optional automatic reminder message deletion

### ISSUE THREAD

**Issue by leonheess** [standup-raven/standup-raven#125]
**Is your feature request related to a problem? Please describe.**
The Raven sends multiple messages reminding people to fill in the stand-up which clutter the chat and have no long-term use.

**Describe the solution you'd like**
A setting so the Raven automatically deletes all reminder messages of the day after sending the stand-up report of the day.

**Comment by harshilsharma63** [comment]
@leonheess that's a great idea! This is fairly easy to implement. Would you like to contribute by implementing this?

**Comment by leonheess** [comment]
@harshilsharma63 Sure, I'll try

**Comment by leonheess** [comment]
@harshilsharma63 I have never used Go before so here is my (probably shitty) attempt: https://github.com/leonheess/standup-raven/tree/GH-125

Not sure how I can test locally either... also `make test` results in:

```
make test
cd server && go get github.com/Masterminds/glide
cd server && C:\Users\Redacted/bin/glide install
/bin/sh: C:UsersRedacted/bin/glide: No such file or directory
make: *** [vendor] Error 127

```

**Comment by harshilsharma63** [comment]
@leonheess hey, you did it almost perfectly! Kudos on your first Go code!

There is just one change needed. Currently, you store all post IDs to be deleted in an in-memory array. This will have side effects when Mattermost is run in high availability mode. I'd recommend storing this information in the provided `KVStore`. But we cannot just keep a single array with KVStore as it will again cause race conditions when multiple Mattermost instances are running.

The plugin is currently not tested to run in HA mode, but I'd like it if incompatibilities are kept to a minimum. Can you think of an idea?

**Comment by leonheess** [comment]
@harshilsharma63 I tried to improve my attempt. As I am not able to set up the project properly I can't test anything or even execute any code, I just coded what I thought would make sense and committed.

I also don't understand your last question. Sorry :/

**Comment by harshilsharma63** [comment]
@leonheess your changes look great!

> The plugin is currently not tested to run in HA mode, but I'd like it if incompatibilities are kept to a minimum. Can you think of an idea?

You can run Mattermost in a high availability mode, as a cluster. So with plugins, a copy of plugin runs on each node in this cluster. Let's say we have standup raven on two nodes trying to process standup for the same channel at the same time. Both nodes would be doing `KVGet` and `KVSet` together, thus causing a race condition.

But I see you've kept a separate key-value entry for each channel, which is sufficiently good for now.

Regarding setting up the project, I'm more than happy to help with that. What issue are you facing? Have you setup Mattermost? Just in case, here are the setup links-

Setting up Standup Raven development - https://github.com/standup-raven/standup-raven/blob/master/docs/getting_started.md
Installing Mattermost - https://docs.mattermost.com/guides/administrator.html#installing-mattermost

**Comment by harshilsharma63** [comment]
Also, can you create a pull request so I can easily see your changes?

**Comment by leonheess** [comment]
Thanks! I will look into those tomorrow (:

**Comment by leonheess** [comment]
Thanks for finishing the work! I unfortunately haven't had the time to revisit this in the past few weeks :/

**Comment by harshilsharma63** [comment]
@leonheess hey, no problem and thanks for the base logic you wrote!

**Comment by harshilsharma63** [comment]
@ThiefMaster this has been implemented in the latest release - 2.3.0. Please update your plugin installation.
